File: src/services/groundstations.ts
// Commercial + agency ground-station networks used for satellite TT&C and
// payload downlink. Public reference data (site locations, operators, bands);
// nothing sensitive. Bands and elevation masks are representative, not
// operator-published values, and only feed the pass/link-budget estimates.
import type { GroundStation } from '../types'

const RAW: Omit<GroundStation, 'id'>[] = [
  // KSAT — Kongsberg Satellite Services (global polar + mid-latitude network)
  {
    name: 'Svalbard (SvalSat)',
    operator: 'KSAT',
    lat: 78.23,
    lng: 15.39,
    bands: ['S', 'X', 'Ka'],
    minElevDeg: 5,
  },
  { name: 'TrollSat', operator: 'KSAT', lat: -72.01, lng: 2.53, bands: ['S', 'X'], minElevDeg: 5 },
  { name: 'Tromsø', operator: 'KSAT', lat: 69.66, lng: 18.94, bands: ['S', 'X'], minElevDeg: 5 },
  {
    name: 'Punta Arenas',
    operator: 'KSAT',
    lat: -52.94,
    lng: -70.85,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'Hartebeesthoek',
    operator: 'KSAT',
    lat: -25.89,
    lng: 27.69,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  { name: 'Mauritius', operator: 'KSAT', lat: -20.31, lng: 57.5, bands: ['S', 'X'], minElevDeg: 5 },
  { name: 'Singapore', operator: 'KSAT', lat: 1.39, lng: 103.9, bands: ['S', 'X'], minElevDeg: 7 },
  { name: 'Awarua', operator: 'KSAT', lat: -46.53, lng: 168.38, bands: ['S', 'X'], minElevDeg: 5 },
  { name: 'Inuvik', operator: 'KSAT', lat: 68.32, lng: -133.55, bands: ['S', 'X'], minElevDeg: 5 },

  // Leaf Space — Leaf Line GSaaS
  {
    name: 'Leaf · Awarua',
    operator: 'Leaf Space',
    lat: -46.52,
    lng: 168.39,
    bands: ['S', 'X'],
    minElevDeg: 8,
  },
  {
    name: 'Leaf · Cordoba',
    operator: 'Leaf Space',
    lat: -31.52,
    lng: -64.46,
    bands: ['S', 'X'],
    minElevDeg: 8,
  },
  {
    name: 'Leaf · Azores',
    operator: 'Leaf Space',
    lat: 37.74,
    lng: -25.66,
    bands: ['S', 'X'],
    minElevDeg: 8,
  },
  {
    name: 'Leaf · Svalbard',
    operator: 'Leaf Space',
    lat: 78.22,
    lng: 15.65,
    bands: ['S', 'X'],
    minElevDeg: 8,
  },

  // RBC Signals
  {
    name: 'RBC · Fairbanks',
    operator: 'RBC Signals',
    lat: 64.86,
    lng: -147.85,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'RBC · Punta Arenas',
    operator: 'RBC Signals',
    lat: -53.0,
    lng: -70.86,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'RBC · Brewster',
    operator: 'RBC Signals',
    lat: 48.2,
    lng: -119.68,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },

  // AWS Ground Station
  {
    name: 'AWS · Ohio',
    operator: 'AWS GS',
    lat: 39.96,
    lng: -83.0,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'AWS · Oregon',
    operator: 'AWS GS',
    lat: 45.84,
    lng: -119.7,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'AWS · Stockholm',
    operator: 'AWS GS',
    lat: 59.33,
    lng: 18.07,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'AWS · Bahrain',
    operator: 'AWS GS',
    lat: 26.07,
    lng: 50.56,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },

  // NASA Near Earth Network
  {
    name: 'Wallops (WGS)',
    operator: 'NASA NEN',
    lat: 37.95,
    lng: -75.46,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'Fairbanks (ASF)',
    operator: 'NASA NEN',
    lat: 64.97,
    lng: -147.51,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'McMurdo (MG1)',
    operator: 'NASA NEN',
    lat: -77.84,
    lng: 166.67,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'White Sands',
    operator: 'NASA NEN',
    lat: 32.5,
    lng: -106.61,
    bands: ['S', 'Ku'],
    minElevDeg: 5,
  },

  // ESA Estrack (deep-space + LEO support)
  {
    name: 'Kiruna (KIR)',
    operator: 'ESA Estrack',
    lat: 67.86,
    lng: 21.06,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'Kourou (KRU)',
    operator: 'ESA Estrack',
    lat: 5.25,
    lng: -52.8,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },
  {
    name: 'New Norcia (NNO)',
    operator: 'ESA Estrack',
    lat: -31.05,
    lng: 116.19,
    bands: ['S', 'X'],
    minElevDeg: 5,
  },

  // Atlas Space Operations — Freedom network
  {
    name: 'Atlas · Guam',
    operator: 'Atlas',
    lat: 13.59,
    lng: 144.86,
    bands: ['S', 'X'],
    minElevDeg: 8,
  },
  {
    name: 'Atlas · Ghana',
    operator: 'Atlas',
    lat: 5.66,
    lng: -0.18,
    bands: ['S', 'X'],
    minElevDeg: 8,
  },
]

export const GROUND_STATIONS: GroundStation[] = RAW.map((s) => ({
  ...s,
  id: `gs-${s.operator.replace(/\W+/g, '').toLowerCase()}-${s.name.replace(/\W+/g, '').toLowerCase()}`,
}))

export const GROUND_OPERATORS = [...new Set(GROUND_STATIONS.map((s) => s.operator))]

File: backend/app/orbital/groundstations.py
"""Ground-station network (mirrors the frontend reference set)."""

from __future__ import annotations

import re
from dataclasses import dataclass, field


@dataclass(frozen=True)
class GroundStation:
    name: str
    operator: str
    lat: float
    lng: float
    bands: tuple[str, ...]
    min_elev_deg: float
    id: str = field(default="")

    def with_id(self) -> GroundStation:
        slug_op = re.sub(r"\W+", "", self.operator).lower()
        slug_name = re.sub(r"\W+", "", self.name).lower()
        return GroundStation(
            name=self.name,
            operator=self.operator,
            lat=self.lat,
            lng=self.lng,
            bands=self.bands,
            min_elev_deg=self.min_elev_deg,
            id=f"gs-{slug_op}-{slug_name}",
        )


_RAW: list[GroundStation] = [
    GroundStation("Svalbard (SvalSat)", "KSAT", 78.23, 15.39, ("S", "X", "Ka"), 5),
    GroundStation("TrollSat", "KSAT", -72.01, 2.53, ("S", "X"), 5),
    GroundStation("Tromso", "KSAT", 69.66, 18.94, ("S", "X"), 5),
    GroundStation("Punta Arenas", "KSAT", -52.94, -70.85, ("S", "X"), 5),
    GroundStation("Hartebeesthoek", "KSAT", -25.89, 27.69, ("S", "X"), 5),
    GroundStation("Mauritius", "KSAT", -20.31, 57.5, ("S", "X"), 5),
    GroundStation("Singapore", "KSAT", 1.39, 103.9, ("S", "X"), 7),
    GroundStation("Awarua", "KSAT", -46.53, 168.38, ("S", "X"), 5),
    GroundStation("Inuvik", "KSAT", 68.32, -133.55, ("S", "X"), 5),
    GroundStation("Leaf - Awarua", "Leaf Space", -46.52, 168.39, ("S", "X"), 8),
    GroundStation("Leaf - Cordoba", "Leaf Space", -31.52, -64.46, ("S", "X"), 8),
    GroundStation("Leaf - Azores", "Leaf Space", 37.74, -25.66, ("S", "X"), 8),
    GroundStation("Leaf - Svalbard", "Leaf Space", 78.22, 15.65, ("S", "X"), 8),
    GroundStation("RBC - Fairbanks", "RBC Signals", 64.86, -147.85, ("S", "X"), 5),
    GroundStation("RBC - Punta Arenas", "RBC Signals", -53.0, -70.86, ("S", "X"), 5),
    GroundStation("RBC - Brewster", "RBC Signals", 48.2, -119.68, ("S", "X"), 5),
    GroundStation("AWS - Ohio", "AWS GS", 39.96, -83.0, ("S", "X"), 5),
    GroundStation("AWS - Oregon", "AWS GS", 45.84, -119.7, ("S", "X"), 5),
    GroundStation("AWS - Stockholm", "AWS GS", 59.33, 18.07, ("S", "X"), 5),
    GroundStation("AWS - Bahrain", "AWS GS", 26.07, 50.56, ("S", "X"), 5),
    GroundStation("Wallops (WGS)", "NASA NEN", 37.95, -75.46, ("S", "X"), 5),
    GroundStation("Fairbanks (ASF)", "NASA NEN", 64.97, -147.51, ("S", "X"), 5),
    GroundStation("McMurdo (MG1)", "NASA NEN", -77.84, 166.67, ("S", "X"), 5),
    GroundStation("White Sands", "NASA NEN", 32.5, -106.61, ("S", "Ku"), 5),
    GroundStation("Kiruna (KIR)", "ESA Estrack", 67.86, 21.06, ("S", "X"), 5),
    GroundStation("Kourou (KRU)", "ESA Estrack", 5.25, -52.8, ("S", "X"), 5),
    GroundStation("New Norcia (NNO)", "ESA Estrack", -31.05, 116.19, ("S", "X"), 5),
    GroundStation("Atlas - Guam", "Atlas", 13.59, 144.86, ("S", "X"), 8),
    GroundStation("Atlas - Ghana", "Atlas", 5.66, -0.18, ("S", "X"), 8),
]

GROUND_STATIONS: list[GroundStation] = [s.with_id() for s in _RAW]
GROUND_OPERATORS: list[str] = list(dict.fromkeys(s.operator for s in GROUND_STATIONS))
